e2e/src/app.e2e-spec.ts

```diff
@@ -0,0 +1,23 @@
+import {MaterialDocsAppPage} from './app.po';
+import {browser, logging} from 'protractor';
+
+describe('Material Docs App', () => {
+  let page: MaterialDocsAppPage;
+
+  beforeEach(() => {
+    page = new MaterialDocsAppPage();
+  });
+
+  it('should display welcome message', async() => {
+    await page.navigateTo();
+    expect(await page.getTitleText()).toEqual('Angular Material');
+  });
+
+  afterEach(async () => {
+    // Assert that there are no errors emitted from the browser
+    const logs = await browser.manage().logs().get(logging.Type.BROWSER);
+    expect(logs).not.toContain(jasmine.objectContaining({
```

Comment:
I'm not sure whether this will actually fail the test. A long time ago I tried doing something similar to detect leaking elements in unit tests and I couldn't get it to fail properly, although Jasmine could've changed the behavior since then. It might be better to either throw an error here or pull it out into a function that is called at the end of each test.

Reply:
This is taken directly from what the Angular CLI's e2e schematic generates for a new project.

If there is a little more concrete guidance here, I could open up an issue against Angular CLI.